src/raw_node.rs

```diff
@@ -204,6 +258,15 @@ pub struct RawNode<T: Storage> {
     pub raft: Raft<T>,
     prev_ss: SoftState,
     prev_hs: HardState,
+    // Current max number of Record and ReadyRecord.
+    max_number: u64,
+    records: VecDeque<ReadyRecord>,
+    // If there is a pending snapshot.
+    pending_snapshot: bool,
```

Comment:
Maybe this flag is unnecessary, IMO, pending snapshot just like pending entries that both waiting for persisting, 
we can handle pending snapshot like how we handle pending entries (stash messages and  `on_persist`-like function)

Reply:
If this peer needs to apply snapshot, it must not belong to quorum members of this raft group.
And the leader will not send the subsequent entries unless there is a new leader and the snapshot has exactly the last entries which the new leader knows, obviously, this case is very rare. So I don't see any advantage to get a `Ready` before applying snapshot.

Reply:
> If this peer needs to apply snapshot, it must not belong to quorum members of this raft group.

I don't think this is true, for example, a raft group with three voters and one follower is down and another is applying snapshot

> So I don't see any advantage to get a Ready before applying snapshot.

The biggest benefit to remove `pending_snapshot` is to reuse the logic of handling pending entries